### DISCUSSION TITLE
use a file on google drive as ics url

### DISCUSSION THREAD

**Discussion by sadrov** [derekantrican/GAS-ICS-Sync#446] [Q&A] [answered]
Hey everyone, 
first of all thanks for this cool script! It's working as expected but now I'm trying to pull the calendar data from an ics file I have stored on google drive (the same google account that is used to run this apps script).
So for ["icsUrl1", "targetCalendar1"] I'm trying to put  ["https://drive.google.com/file/d/xxxxxxxxxxxxxxxxxxxx", "my_calendar"]. Of course x's are replaced by the actual file id. 
The script then outputs: 
[ERROR] Incorrect ics/ical URL: https://drive.google.com/file/d/xxxxxxxxxxxxxxxxxxxx 
even though this file is an ics file that syncs just fine when I pull it from another url (that is not google drive).
Has anyone tried something similar or know what the issue is here? The error is the same wether the file is set to private or public.

Thanks in advance for your ideas/insights/replies!

**Comment by jonas0b1011001** [comment, marked as answer]
These Google Drive links do not directly serve the file content.
This should help #412

Given the url format you'd have to adjust the regex to get the file id
`const IDregex = /https:\/\/drive\.google\.com\/file\/d\/([^\/]+)\/?/;`

You will have to reauthorize the script to get access to your Drive files.

**Reply by sadrov** [reply]
Yep, that totally worked with this adjusted regex. Thanks a lot for your help @jonas0b1011001  !
